nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,323 @@
+
+#### Filter Logic
+
++/- 1.4.0
+    Custom field [extended filtering](../rest-api/filtering.md#lookup-expressions) introduced extended lookup expression filters for `exact` and `icontains`, duplicating the functionality of both the `Strict` and `Loose` settings.
+
+The filter logic setting applies to filtering on custom fields in the UI and API. For example, when filtering in the API to find a device with the custom field `cf1` set to `"abc"` you would query `/api/dcim/devices/?cf_cf1=abc`. If the filter logic setting is set to `Loose` this would match on `"ABC"` and `"abcdef"`. If the filter logic setting is set to `Strict` only devices with the custom field set to exactly "abc" (case sensitive) would be returned. If the filter logic setting is set to `disabled`, no filters will be available for this custom field, including extended lookup filters. The `Loose` and `Strict` settings only change the behavior of the default filter (`cf_customfieldname`) on `text`, `url` and `json` custom fields.
+
+#### Move to Advanced Tab
+
+When selected, the custom field will appear in the "Advanced" tab of the object detail view instead of the default tab.
+
+### Assignment
+
+#### Content Types
+
+The list of content types to add this custom field to. Only models that subclass the `nautobot.extras.models.customfields.CustomFieldModel` model class can be selected.
+
+!!! note
+    When a custom field is created or associated to a new content type (model), all affected existing objects will be updated to add the custom field. The initial value will be set to the `default` value of the custom field. This update runs as a background task via [Celery](../installation/services.md#worker-service), so it may take a few seconds or more before all objects reflect the new custom field, depending on the size of your database.
+
+### Validation Rules
+
+Validation rules are used for constraining custom fields to specific values.
+
+#### Minimum value
+
+Minimum allowed value for `Integer` fields.
+
+#### Maximum value
+
+Maximum allowed value for `Integer` fields.
+
+#### Validation Regex
+
+Regular expression to enforce on `Text`, `URL`, `Selection` and `Multiple selection` field values. Regex validation is handled by the [python re engine](https://docs.python.org/3/library/re.html) which uses a PCRE or perl-like regular expression syntax. Examples of common regex validations:
+
+Must start with companyname
+> ^companyname
+
+Must end with 5 digit zip code
+> [0-9]{5}$
+
+Must only contain digits
+> ^\d+$
+
+Must be exactly 8 alphanumeric characters
+> ^[0-9a-zA-Z]{8}$
+
+Must be between 8 and 10 alphanumeric characters and underscore
+> ^\w+{8,10}$
+
+Must contain anything that is not whitespace
+> \S
```

Comment:
IMO change these example regex to code gates with `no-highlight` vs. block quote.